packages/components/addon/components/hds/side-nav/section/link.hbs

```diff
@@ -0,0 +1,31 @@
+<li class="hds-side-nav__list-item">
+  <Hds::Interactive
+    @href={{@href}}
+    @route={{@route}}
+    @models={{hds-link-to-models @model @models}}
+    @query={{hds-link-to-query @query}}
+    @icon={{@icon}}
+    @isRouteExternal={{@isRouteExternal}}
+    @current-when={{@current-when}}
+    class="hds-side-nav__item hds-side-nav__item--link"
+    ...attributes
+  >
+    {{#if @icon}}
+      <FlightIcon class="hds-side-nav__list-icon" @name={{@icon}} @stretched={{true}} />
+    {{/if}}
+
+    <span class="hds-side-nav-label hds-typography-body-200 hds-font-weight-medium">
+      {{@label}}
+    </span>
+
+    {{#if @count}}
+      <Hds::BadgeCount @text={{@count}} @type="inverted" @size="small" />
+    {{/if}}
+
+    {{#if @badge}}
+      <Hds::Badge @text={{@badge}} @type="inverted" @size="small" />
+    {{/if}}
+
+    {{yield}}
```

Comment:
@KristinLBradley I've just noticed that this component is missing one icon:
<img width="305" alt="modified" src="https://user-images.githubusercontent.com/686239/221200796-f4077a41-517b-4b79-83b3-389417507450.png">

Reply:
I hadn't looked at how to handle nested link related elements yet. I can add that. It seems like it should be handled with an extra argument? I'll look at it.

Reply:
The component itself (the `::Link` I mean) doesn't know anything about it being nested or not, is just how it's used (for nesting). And yes, it's just a matter of adding another argument (name TBD) and make sure the layout still works.

Reply:
I understand. I'll add that next.